File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int ans =0;
    int goodNodes(TreeNode* root) {
        if (root==NULL) return 0;
        preorder(root,root->val);
        return ans;
    }
    void preorder(TreeNode* root, int maxVal){
        if (root==NULL) return ;
        if (root->val>=maxVal) ans++;
        preorder(root->left,max(maxVal,root->val));
        preorder(root->right,max(maxVal,root->val));
    }
};

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    ans = 0
    def goodNodes(self, root: TreeNode) -> int:
        
        self.ans = 0
        
        def count(node,prevbest):
            if node==None:
                return 
            
            
            if node.val>=prevbest:
                self.ans += 1
            best = max (node.val,prevbest)
            
            if node.left:
                count(node.left,best)
                
            if node.right:
                count(node.right,best)
                
                
                
        count(root,float('-inf'))
        return self.ans